bemserver_ui/static/scripts/modules/views/analysis/weather.js

```diff
@@ -0,0 +1,330 @@
+                    `api.analysis.weather.retrieve`,
+                    {
+                        site_id: this.#structuralElementId,
+                        period_type: this.#periodTypeSelectElmt.value,
+                        period_day: this.#periodDaySelectElmt.value,
+                        period_month: this.#periodMonthSelectElmt.value,
+                        period_year: this.#periodYearSelectElmt.value,
+                        timezone: this.#tzName,
+                        forecast: this.#forecast.checked,
+                    }
+                ),
+                (data) => {
+                    this.#mainChartContainerElmt.innerHTML = "";
+                    if ( data == null || data.length == 0) {
+                        let colElmt = document.createElement("div");
+                        colElmt.classList.add("col", "text-start", "text-muted");
+
+                        let pNoDataElmt = document.createElement("p");
+                        pNoDataElmt.classList.add("fst-italic");
+                        pNoDataElmt.innerText = "No data available";
+                        colElmt.appendChild(pNoDataElmt);
+
+                        let pHelpElmt = document.createElement("p");
+                        pHelpElmt.innerText = `Maybe the view is not configured for this ${this.#structuralElementType}.`;
+                        colElmt.appendChild(pHelpElmt);
+
+                        if (!signedUser.is_admin) {
+                            pHelpElmt.classList.add("mb-0");
+
+                            let pHelpNotAdminElmt = document.createElement("p");
+                            pHelpNotAdminElmt.innerText = "Please contact your campaign administrator.";
+                            colElmt.appendChild(pHelpNotAdminElmt);
+                        }
+
+                        this.#mainChartContainerElmt.appendChild(colElmt);
+                    }
+                    else {
+                        for (let [name, dataset] of Object.entries(data)) {
+                            let weatherChart = new TimeseriesChartWeather();
+
+                            this.#chartWeather[name] = weatherChart;
+
+                            let chartContainerElmt = document.createElement("div");
+                            chartContainerElmt.classList.add("border", "border-1", "rounded", "justify-content-center", "bg-white", "p-2");
+                            chartContainerElmt.appendChild(weatherChart);
+
+                            let colElmt = document.createElement("div");
+                            colElmt.classList.add("col");
+                            colElmt.appendChild(chartContainerElmt);
+
+                            this.#mainChartContainerElmt.appendChild(colElmt);
+
+                            weatherChart.showLoading();
+                            weatherChart.load(name, dataset, this.#timeFormatPerPeriodType[this.#periodTypeSelectElmt.value], this.#forecast.checked, signedUser.is_admin);
```

Comment:
When calling `weatherChart.load(...)` add the callback that updates modal content on ts info chart toolbox feature.
Something like:
`...load(..., () => { this.#updateTsInfoModal(name, dataset...); });`